apps/astarte_data_updater_plant/lib/astarte_data_updater_plant/amqp_triggers_producer.ex

```diff
@@ -28,23 +28,30 @@ defmodule Astarte.DataUpdaterPlant.AMQPTriggersProducer do
   """
 
   require Logger
-  use GenServer
+  use GenServer, restart: :transient
 
-  alias AMQP.Channel
+  alias AMQP.{Channel, Connection}
   alias Astarte.DataUpdaterPlant.Config
 
-  @connection_backoff 10000
+  @connection_backoff 10_000
   @adapter Config.amqp_adapter!()
 
-  # Publisc API
-
   @doc """
     Starts the AMQP triggers producer GenServer.
     It initializes the connection to the AMQP broker and prepares it for publishing messages.
     The `args` parameter can be used to pass additional configuration options if needed.
   """
   def start_link(args \\ []) do
-    GenServer.start_link(__MODULE__, args, name: __MODULE__)
+    realm_name = Keyword.fetch!(args, :realm)
+
+    Logger.info("AMQPTriggers producer start link for realm #{realm_name}",
+      tag: "amqp_triggers_consumer_start_link"
+    )
+
+    with {:error, {:already_started, pid}} <-
+           GenServer.start_link(__MODULE__, args, name: server) do
+      {:ok, pid}
```

Comment:
this is now handled at the supervisor level so it can be simplified here

```suggestion
    GenServer.start_link(__MODULE__, args, name: server)
```